Workers/DatabaseSyncWorker/DatabaseSyncLogic.cs

```diff
@@ -64,6 +71,48 @@ public DatabaseSyncLogic(PostgresOptions postgresOptions, WorkerConnection conne
 
         public void ProcessOpList(OpList opList)
         {
+            whenWorker.ProcessOpList(opList);
+            whenWorkerClient.ProcessOpList(opList);
+
+            foreach (var workerEntityId in whenWorker.Deactivated)
+            {
+                if(clients.TryGet(workerEntityId, out var client))
+                {
+                    clientWorkers.TryRemove(client.PlayerIdentity.PlayerIdentifier, out var removed);
```

Comment:
Do we expect this to ever return `false`? If so, `removed` could be `null`.

Reply:
Similarly the log inside the `else if` below doesn't make sense if the `TryRemove` return `false`

Reply:
`Try*` is the only API available for concurrent collections, will flesh this out more